lib/index.js

```diff
@@ -92,64 +103,145 @@ Rest.prototype = {
     } else if(_.isUndefined(options)) {
       options = {};
     };
-    
-    var controller = this._load(name);
+
+    var versions = options.version || options.versions || ['*'];
+    if(!Array.isArray(versions) ) { versions = [versions] };
+
+    var controller = this._load(name, versions);
 
     if(this.subfolder) {
       options.nestedPath = this.nestedPath;
     }
+    var resources = new Resources(this, name, options);
 
-    var resources = new Resources(this, name, _.merge(controller.options, options));
-  
     resources._init(controller);
 
     if('function' == typeof callback) {
       resources.parent = this;
       resources._nest(callback);
     }
-    
+
     this.addRoutes(resources.routes);
 
     return this;
   },
   mountRoutes: function(app) {
     var that = this;
-    //map the routes
     this.routes.forEach(function(route) {
       that._createRoute(route, app);
     });
-    
   },
-  _load: function(name) {
-    this._loaded = this._loaded || {};
+  _load: function(name, versions) {
+    var that = this;
+    var loaded = {};
+    var base_dir = that.settings.controllers;
+
+    _.each(versions, function(version) {
+      //Search in non searched versions plus wild card version
+      var _versions = _.clone(versions);
+      if ( _versions.indexOf("*") === -1 ) { _versions.unshift("*") }
+      _versions = _versions.splice(0, _versions.indexOf(version) + 1);
+
+      //Loop until find a folder
+      while( _versions.length > 0 ) {
+        var _version = _versions.pop();
+        var version_folder = ""
+        if(_version !== "*") {
+          version_folder = "/v" + _version;
+        }
+
+        var dir = null;
+        if(version !== "") {
+          dir = base_dir + version_folder;
+        } else {
+          dir = base_dir;
+        }
+
+        if(that.subfolder) {
+          dir = path.join(dir, that.subfolder);
+        }
+        _path = path.join(dir, name) + ".js";
+
+        if( fs.existsSync(_path) ) {
+          loaded[version] = require(_path);
+          break;
+        }
+      };
+      if (_.isUndefined( loaded[version] ) ) {
+        return res.status(error).end();
```

Comment:
:warning: ```res``` is not defined. The _load function is called when the app starts.
I think you should throw an exception or an warning.
What do you think?